front-end-tools/src/Main.tsx

```diff
@@ -1081,19 +1445,974 @@ export default function Main(props: ConnectionProps) {
+                                {smartContractIndexError !== undefined && (
                                     <div className="alert alert-danger" role="alert">
                                         Error: {smartContractIndexError}.
                                     </div>
                                 )}
-                                {smartContractIndex !== '' && (
+                                {smartContractIndex !== undefined && (
                                     <div className="actionResultBox">
                                         Smart Contract Inedex:
                                         <div>{smartContractIndex}</div>
                                     </div>
                                 )}
-                            </TestBox>
+                            </Box>
+
+                            {/* Additional helpers: */}
+
+                            <Form className="Box">
+                                <Row>
+                                    <Form.Group className="col-md-4 mb-3">
+                                        <Form.Label>Smart Contract Index</Form.Label>
+                                        <Form.Control
+                                            defaultValue={1}
+                                            type="number"
+                                            min="0"
+                                            {...attributeForm.register('smartContractIndex', { required: true })}
+                                        />
+                                        <Form.Text />
+                                        {attributeForm.formState.errors.smartContractIndex && (
+                                            <Alert key="info" variant="info">
+                                                {' '}
+                                                Smart contract index is required{' '}
+                                            </Alert>
+                                        )}
+                                    </Form.Group>
+
+                                    {attributeForm.watch('deriveFromSmartContractIndex') &&
+                                    contractInstanceInfo !== undefined &&
+                                    contractInstanceInfo.contractName !== undefined ? (
+                                        <Form.Group className="col-md-4 mb-3">
+                                            <Form.Label>Smart Contract Name</Form.Label>
+                                            <Form.Control
+                                                value={
+                                                    contractInstanceInfo?.contractName
+                                                        ? contractInstanceInfo.contractName
+                                                        : 'undefined'
+                                                }
+                                                disabled
```

Comment:
In terms of styling, I think the `disabled` is more clear since it greys out the field and doesn't allow the user to click into the field (the boarders are not highlighted or focused when clicking into the field). The `read-only`  styling allows the user to click into the field and focus on the field and since the field is not greyed out it looks like as if it could be edited by the user.

Reply:
Hmm.. in bootstrap 4 `readonly` would render style like the disabled of bootstrap 5.

Anyway, as I recall disabled inputs are excluded from the form values (native browser form behaviour, not react-hook-form exclusive), so in my opinion the correct way would still be to mark them as `readonly`.

In the [bootstrap docs for disabled form controls](https://getbootstrap.com/docs/5.3/forms/form-control/#disabled), there is an example marking an input as `disabled readonly`, which makes the text selectable (which is also not possible with just `disabled`. Maybe this will include values in form submissions as well?

Reply:
The idea is to `disable` the input field, so users cannot fill in the the value and submit it via the form. The smart contract name was derived from the smart contract index. It should be most obviously that this field cannot be filled in. `Disabled` styling is pretty clear in that sense by making the field greyed out that is why I would lick to stick with the `disabled` styling.

Reply:
I agree the styling for disabled is as it should be, however I was just trying to make the point that if you rely on the value from the disabled input field on the form submission values, it will not be available:

```ts
const {handleSubmit, register} = useForm();
function onSubmit({field1, field2}) {
    // field1 is defined
    // field2 is undefined
}
...
<form onSubmit={handleSubmit(onSubmit)}>
    <input {...register('field1')} />
    <input {...register('field2')} diasbled />
</form>
```

Does that make sense?

Anyway if you don't use the value from the input field it won't matter.